### PR TITLE
`TextInput`: add support for month, week, tel types

### PR DESCRIPTION
**Pull request by shleewhite**
### :pushpin: Summary

If merged, this PR will add support for the following input types: month, week, tel.

### :hammer_and_wrench: Detailed description

- Updates the input type argument type
- Updates CSS to have month and week input have width='initial'
- Update showcase test error messsage
- Website
  - add new types to prop table
  - add examples of new types to the guidelines

### :camera_flash: Screenshots

**Updated documentation site in Chrome:**
Screenshot 2024-07-16 at 3 52 18 PM

**Updated documentation site in Safari:**
Screenshot 2024-07-16 at 3 45 02 PM

**Added new types to showcase**
Screenshot 2024-07-16 at 4 49 36 PM

### :link: External links

Jira ticket: [HDS-3555](https://hashicorp.atlassian.net/browse/HDS-3555)

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3555]: https://hashicorp.atlassian.net/browse/HDS-3555?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by didoo** [comment]
@shleewhite can you also update the showcase page for the `Form:TextInput`?

- you can see it here https://hds-showcase.vercel.app/components/form/text-input
- how to run it locally is explained here: https://github.com/hashicorp/design-system/blob/main/packages/components/CONTRIBUTING.md#running-the-showcase-application)
- this is the file you need to update: `showcase/app/templates/components/form/text-input.hbs`

Let me know if something about the showcase (or how to use it) is not clear. Thanks

**Comment by shleewhite** [comment]
> @shleewhite can you also update the showcase page for the `Form:TextInput`?
> 
> * you can see it here https://hds-showcase.vercel.app/components/form/text-input
> * how to run it locally is explained here: https://github.com/hashicorp/design-system/blob/main/packages/components/CONTRIBUTING.md#running-the-showcase-application)
> * this is the file you need to update: `showcase/app/templates/components/form/text-input.hbs`
> 
> (by the look of it, one section is automatically updated, because we loop on `TYPES`, while the other is a manual list)
> 
> Let me know if something about the showcase (or how to use it) is not clear. Thanks

Added!

**Review by didoo — APPROVED** [review]
Wow, that was super fast!! 🙌

For me everything is spot on, but please wait for @alex-ju review/approval too, before merging (he's the "go to person" for everything `Hds::Form` related)

File: .changeset/good-pumas-hunt.md
Comment (alex-ju) [inline .changeset/good-pumas-hunt.md:5]: We aim to [use past tense in changelogs](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#tips-for-writing-consistent-changelog-entries) – and yes, I see how that's unnatural compared to git commit messages

```suggestion
`Form::TextInput` - added support for `"month"`, `"week"`, and `"tel"` input types
```

**Review by alex-ju — APPROVED** [review]
Looking good! Left a suggestion for the changelog, otherwise it's good to be merged

**Pull request merged**
